extensions/azdata/src/typings/azdata-ext.d.ts

```diff
@@ -278,13 +279,14 @@ declare module 'azdata-ext' {
 							memoryLimit?: string,
 							memoryRequest?: string,
 							noWait?: boolean,
-						}
+						},
+						additionalEnvVars?: { [key: string]: string }
 					): Promise<AzdataOutput<void>>
 				}
 			}
 		},
 		getPath(): Promise<string>,
-		login(endpoint: string, username: string, password: string): Promise<AzdataOutput<any>>,
+		login(endpoint: string, username: string, password: string, additionalEnvVars?: { [key: string]: string }): Promise<AzdataOutput<any>>,
```

Comment:
are you sure you really need to pass in the config file the cluster context as environment here to login, if i remember correctly, the kubeconfig can be a environment but the cluster context needs to be set in the kubeconfig file before running the login command.

Reply:
Yeah sync up with Sean on this one - login would be just for the controller endpoint I would think. But it's possible it's hitting the k8 cluster for some reason too.

Although in general I'd probably say it's safer to set this regardless in case that changes later on anyways. But it'd be good to know either way.

Reply:
yeah I have discussed with Sean and he is doing his own research as well

Reply:
Sean and co will make this easier in future, for now we will stick with an implementation that works where the login does require the kubeconfig and kube cluster information for now.